fraud/service.go

```diff
@@ -1,23 +1,41 @@
 package fraud
 
 import (
+	"bytes"
 	"context"
 	"encoding/hex"
 	"errors"
 	"fmt"
 	"sync"
-	"time"
 
 	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
 	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
-)
+	"golang.org/x/sync/errgroup"
+
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
 
-const (
-	// fetchHeaderTimeout duration of GetByHeight request to fetch an ExtendedHeader.
-	fetchHeaderTimeout = time.Minute * 2
+	pb "github.com/celestiaorg/celestia-node/fraud/pb"
+	"github.com/celestiaorg/celestia-node/params"
 )
 
+// fraudRequests is amount of external requests that will be done in order to get fraud proofs from another peers.
+const fraudRequests = 5
```

Comment:
have we documented somewhere that 5 is enough?

Reply:
I will update an ADR with all explanations.